app/authentication/user_id_generator.py

```diff
@@ -0,0 +1,30 @@
+from app import settings
+from app.metadata.metadata_store import MetaDataConstants
+from app.utilities.strings import to_str
+import hashlib
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class UserIDGenerator(object):
+
+    @staticmethod
+    def generate_id(token):
+        logger.debug("About to generate a user id")
+        ru_ref = token.get(MetaDataConstants.RU_REF)
+        collection_exercise_sid = token.get(MetaDataConstants.COLLECTION_EXERCISE_SID)
+        eq_id = token.get(MetaDataConstants.EQ_ID)
+
+        logger.debug("Using values %s, %s and %s with a salt", ru_ref, collection_exercise_sid, eq_id)
```

Comment:
Worth adding an assert (maybe even if+exception?) here to ensure these values exist/aren't empty etc?